run.sh

```
@@ -23,14 +23,40 @@
 
 # Cross-platform venv activation and path handling
 VENV_DIR="venv"
+IS_WINDOWS=false
 if [[ "$OSTYPE" == "msys" || "$OSTYPE" == "cygwin" || "$OSTYPE" == "win32" ]]; then
     # Windows (using Git Bash, Cygwin, etc.)
     VENV_ACTIVATE="$VENV_DIR/Scripts/activate"
+    IS_WINDOWS=true
```

Shellcheck (0.11.0)

[warning] IS_WINDOWS appears unused. Verify use (or export if used externally).

(SC2034)